lib/Reader/reader.cpp

```diff
@@ -3816,7 +3812,9 @@ IRNode *ReaderBase::loadObj(CORINFO_RESOLVED_TOKEN *ResolvedToken,
     Class = ResolvedToken->hClass;
     TheCorInfoType = getClassType(Class);
   }
-
+  if (isVectorPointerType(Address)) {
```

Comment:
I think that I'd prefer that this logic were moved into `GenIR::loadNonPrimitiveObj`.